add clarification for maxSubqueryRows (#11687)
apache/druid: docs/configuration/index.md

|`druid.server.http.enableRequestLimit`|If enabled, no requests would be queued in jetty queue and "HTTP 429 Too Many Requests" error response would be sent. |false|
|`druid.server.http.defaultQueryTimeout`|Query timeout in millis, beyond which unfinished queries will be cancelled|300000|
|`druid.server.http.maxScatterGatherBytes`|Maximum number of bytes gathered from data processes such as Historicals and realtime processes to execute a query. Queries that exceed this limit will fail. This is an advance configuration that allows to protect in case Broker is under heavy load and not utilizing the data gathered in memory fast enough and leading to OOMs. This limit can be further reduced at query time using `maxScatterGatherBytes` in the context. Note that having large limit is not necessarily bad if broker is never under heavy concurrent load in which case data gathered is processed quickly and freeing up the memory used. Human-readable format is supported, see [here](human-readable-byte.md). |Long.MAX_VALUE|

> |`druid.server.http.maxSubqueryRows`|Maximum number of rows from subqueries per query. Druid stores the subquery rows in temporary tables that live in the Java heap. When a subquery exceeds this guardrail that prevents the system from exhausting available heap, Druid throws a resource limit exceeded exception: "Subquery generated results beyond maximum."<br>It is a good practice to avoid large subqueries in Druid. However, if you choose to raise the subquery row limit, you must also increase the heap size of all Brokers, Historicals, and task Peons that process data for the subqueries to accommodate the subquery results.<br>There is no formula to calculate the correct value. Trial and error is the best approach.|100000|

Author: ```suggestion
|`druid.server.http.maxSubqueryRows`|Maximum number of rows from subqueries per query. Druid stores the subquery rows in temporary tables that live in the Java heap. This guardrail that prevents the system from exhausting available heap. When a subquery exceeds the row limit, Druid throws a resource limit exceeded exception: "Subquery generated results beyond maximum."<br><br>It is a good practice to avoid large subqueries in Druid. However, if you choose to raise the subquery row limit, you must also increase the heap size of all Brokers, Historicals, and task Peons that process data for the subqueries to accommodate the subquery results.<br><br>There is no formula to calculate the correct value. Trial and error is the best approach.|100000|
```

> |`druid.server.http.maxSubqueryRows`|Maximum number of rows from subqueries per query. Druid stores the subquery rows in temporary tables that live in the Java heap. When a subquery exceeds this guardrail that prevents the system from exhausting available heap, Druid throws a resource limit exceeded exception: "Subquery generated results beyond maximum."<br>It is a good practice to avoid large subqueries in Druid. However, if you choose to raise the subquery row limit, you must also increase the heap size of all Brokers, Historicals, and task Peons that that process data for the subqueries to accommodate the subquery results.<br>There is no formula to calculate the correct value. Trial and error is the best approach.|100000|

Reviewer: A typo (fixed in my suggested text below), but also two other optional suggestions: 
1) To make the `<br>`s look like paragraph breaks requires two of them. (The `druid.access.jdbc.allowedProperties` description in this file, for example, has them as a pair.) Otherwise, they can line up with the start of line, as in github's HTML view of the second `<br>`, which lines up with a natural line start.  
2) I'd split this sentence into two: "When a subquery exceeds this guardrail that prevents the system from exhausting available heap, Druid throws a resource limit exceeded exception":
Something like:
"This property acts as a guardrail that prevents the system from exhausting available heap. If a subquery exceeds this limit, Druid throws a resource limit exceeded exception."

```suggestion
|`druid.server.http.maxSubqueryRows`|Maximum number of rows from subqueries per query. Druid stores the subquery rows in temporary tables that live in the Java heap. When a subquery exceeds this guardrail that prevents the system from exhausting available heap, Druid throws a resource limit exceeded exception: "Subquery generated results beyond maximum."<br>It is a good practice to avoid large subqueries in Druid. However, if you choose to raise the subquery row limit, you must also increase the heap size of all Brokers, Historicals, and task Peons that process data for the subqueries to accommodate the subquery results.<br>There is no formula to calculate the correct value. Trial and error is the best approach.|100000|
```

> |`druid.server.http.maxSubqueryRows`|Maximum number of rows from subqueries per query. Druid stores the subquery rows in temporary tables that live in the Java heap. This guardrail that prevents the system from exhausting available heap. When a subquery exceeds the row limit, Druid throws a resource limit exceeded exception: "Subquery generated results beyond maximum."<br><br>It is a good practice to avoid large subqueries in Druid. However, if you choose to raise the subquery row limit, you must also increase the heap size of all Brokers, Historicals, and task Peons that process data for the subqueries to accommodate the subquery results.<br><br>There is no formula to calculate the correct value. Trial and error is the best approach.|100000|

Author: ```suggestion
|`druid.server.http.maxSubqueryRows`|Maximum number of rows from subqueries per query. Druid stores the subquery rows in temporary tables that live in the Java heap. `druid.server.http.maxSubqueryRows` is a guardrail to prevent the system from exhausting available heap. When a subquery exceeds the row limit, Druid throws a resource limit exceeded exception: "Subquery generated results beyond maximum."<br><br>It is a good practice to avoid large subqueries in Druid. However, if you choose to raise the subquery row limit, you must also increase the heap size of all Brokers, Historicals, and task Peons that process data for the subqueries to accommodate the subquery results.<br><br>There is no formula to calculate the correct value. Trial and error is the best approach.|100000|
```

> |`druid.server.http.maxSubqueryRows`|Maximum number of rows from subqueries per query. Druid stores the subquery rows in temporary tables that live in the Java heap. `druid.server.http.maxSubqueryRows` is a guardrail to prevent the system from exhausting available heap. When a subquery exceeds the row limit, Druid throws a resource limit exceeded exception: "Subquery generated results beyond maximum."<br><br>It is a good practice to avoid large subqueries in Druid. However, if you choose to raise the subquery row limit, you must also increase the heap size of all Brokers, Historicals, and task Peons that process data for the subqueries to accommodate the subquery results.<br><br>There is no formula to calculate the correct value. Trial and error is the best approach.|100000|

Reviewer 2: nit: When I first read this the `per query` limit made me think the limit was for each sub query. Maybe this clears that up slightly. I don't know if this is the best way to articulate it, so feel free to ignore this suggestion
```suggestion
|`druid.server.http.maxSubqueryRows`|Maximum number of rows from all subqueries per query. Druid stores the subquery rows in temporary tables that live in the Java heap. `druid.server.http.maxSubqueryRows` is a guardrail to prevent the system from exhausting available heap. When a subquery exceeds the row limit, Druid throws a resource limit exceeded exception: "Subquery generated results beyond maximum."<br><br>It is a good practice to avoid large subqueries in Druid. However, if you choose to raise the subquery row limit, you must also increase the heap size of all Brokers, Historicals, and task Peons that process data for the subqueries to accommodate the subquery results.<br><br>There is no formula to calculate the correct value. Trial and error is the best approach.|100000|
```

Author: Good clarification, @suneet-s . Thank you!
